src/importlinter/domain/imports.py

```diff
@@ -60,29 +63,29 @@ def is_package(self) -> bool:
         raise NotImplementedError
 
 
-class DirectImport(ValueObject):
+class ImportExpression(ValueObject):
     """
-    An import between one module and another.
+    A python import expression.
     """
 
-    def __init__(
-        self,
-        *,
-        importer: Module,
-        imported: Module,
-        line_number: Optional[int] = None,
-        line_contents: Optional[str] = None,
-    ) -> None:
+    def __init__(self, importer: str, imported: str) -> None:
         self.importer = importer
         self.imported = imported
-        self.line_number = line_number
-        self.line_contents = line_contents
+
+    def to_modules(self, graph: ImportGraph) -> Iterable[Tuple[Module, Module]]:
+        importer = []
+        imported = []
+
+        for module in graph.modules:
```

Comment:
Performance improvement: we only need to look through the entire graph if the import expression contains a wildcard - otherwise we can convert it to a `DirectImport` much more cheaply.

Reply:
Any suggestion how to do this in a nice way? Just check if expression contains one if "*?["?

Reply:
see your comment below